public/app/features/alerting/unified/components/rules/state-history/useRuleHistoryRecords.tsx

```diff
@@ -69,6 +71,46 @@ export function useRuleHistoryRecords(stateHistory?: DataFrameJSON, filter?: str
   }, [stateHistory, filter, theme]);
 }
 
+export function getRuleHistoryRecordsForPanel(stateHistory?: DataFrameJSON, filter?: string) {
```

Comment:
Could we move this to `state-history/common.ts`? Is this function anywhere invoked with the `filter` parameter?
If not, perhaps we could simplify it and remove the code responsible for handling filtering

Reply:
💯